File: src/main/java/nidhal/stackoverflowstatus/entity/Question.java
package nidhal.stackoverflowstatus.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Question {
    @Id
    private long question_id;
    public boolean is_answered;
    // Adjust the length of the column to avoid the following error:
    // Data truncation: Data too long for column 'title' at row 1
    @Column(length = 510)
    private String title;
    @Column(length = 510)
    private String link;
    private int score;
    private int answer_count;
    public long accepted_answer_id;
    private int view_count;
    @ElementCollection
    private List<String> tags;
    private long creation_date;
    public long closed_date;
    public long last_activity_date;
    public long last_edit_date;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "owner_id")
    private Owner owner;
}

File: backend/dotnet/stackoverflow_statistics/Models/Question.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;

namespace stackoverflow_statistics.Models;

public class Question
{
    [Key]
    [JsonProperty("question_id")]
    public long QuestionId { get; init; }

    [JsonProperty("is_answered")]
    public bool IsAnswered { get; init; }

    [MaxLength(510)]
    [JsonProperty("title")]
    public string Title { get; init; }

    [MaxLength(510)]
    [JsonProperty("link")]
    public string Link { get; init; }

    [JsonProperty("score")]
    public int Score { get; init; }

    [JsonProperty("answer_count")]
    public int AnswerCount { get; init; }

    [JsonProperty("accepted_answer_id")]
    public long AcceptedAnswerId { get; init; }

    [JsonProperty("view_count")]
    public int ViewCount { get; init; }

    [JsonProperty("creation_date")]
    public long CreationDate { get; init; }

    [JsonProperty("closed_date")]
    public long ClosedDate { get; init; }

    [JsonProperty("last_activity_date")]
    public long LastActivityDate { get; init; }

    [JsonProperty("last_edit_date")]
    public long LastEditDate { get; init; }

    [NotMapped]
    [JsonProperty("tags")]
    public List<string> Tags { get; init; }
}